### ISSUE TITLE
Crashes after update from 3.1.4 to 4.1.3

### ISSUE THREAD

**Issue by kirill-21** [dart-windows/win32#705]
```dart
import 'dart:ffi';
import 'dart:io';

import 'package:ffi/ffi.dart';
import 'package:win32/win32.dart';

main() async {
  await getDesktopFolders();
  print('213');

  await Future.delayed(const Duration(seconds: 10));
  print('21');
}

/// Retrieves all user's desktop paths.
/// Example: {C:\Users\Test\Desktop, C:\Users\Test\OneDriveDesktop}
Future<Set<String>> getDesktopFolders() async {
  // Final value to be filled and returned
  final Set<String> directories = {};

  // Create pointers for 3 different WinAPI methods of retrieving known folders
  final Pointer<GUID> appsFolder = GUIDFromString(FOLDERID_Desktop);
  final Pointer<PWSTR> ppszPath = calloc<PWSTR>();
  final Pointer<COMObject> ppkf = calloc<COMObject>();
  final Pointer<Utf16> path = calloc<WCHAR>(MAX_PATH).cast();

  // Try modern WinAPI way of retrieving the desktop folder
  try {
    // Fill ppszPath pointer with possible desktop path
    final int _result = SHGetKnownFolderPath(
      appsFolder,
      KF_FLAG_DEFAULT,
      NULL,
      ppszPath,
    );

    // Stop function if SHGetKnownFolderPath was not completed with
    // successful result
    if (FAILED(_result)) throw 'SHGetKnownFolderPath method has failed';

    // Check if retrieved WinAPI directory exists
    final String _winAPIPath = ppszPath.value.toDartString();
    if (await Directory(_winAPIPath).exists()) {
      directories.add(_winAPIPath.toLowerCase());
    }
  } catch (_) {
    /* Do nothing */
  }

  // Get the path for a known Winodws folder, using the COM API
  try {
    CoInitializeEx(nullptr, COINIT_APARTMENTTHREADED);

    // Get instance of KnownFolderManager
    final KnownFolderManager knownManager = KnownFolderManager.createInstance();
    int hr = knownManager.getFolder(appsFolder, ppkf.cast());
    if (FAILED(hr)) throw 'KnownFolderManager method has failed';

    final knownFolder = IKnownFolder(ppkf);
    hr = knownFolder.getPath(0, ppszPath);
    if (FAILED(hr)) throw 'IKnownFolder method has failed';

    // Convert retrieved path to dart string, uninitialize no longer needed
    // KnownFolderManager instance.
    final path = ppszPath.value.toDartString();
    CoUninitialize();
    if (await Directory(path).exists()) directories.add(path.toLowerCase());
  } catch (_) {
    /* Do nothing */
  }

  // Try deprecated windows api (still works better than new APIs on old
  // windows version)
  try {
    final result = SHGetFolderPath(
      NULL,
      CSIDL_DESKTOP,
      NULL,
      0,
      path,
    );

    if (SUCCEEDED(result)) {
      final String dartPath = path.toDartString();
      if (await Directory(dartPath).exists()) {
        directories.add(dartPath.toLowerCase());
      }
    }
  } catch (_) {
    /* Do nothing */
  }

  calloc.free(appsFolder);
  calloc.free(ppkf);
  calloc.free(ppszPath);
  calloc.free(path);

  // Return default path if none of above exist
  return directories;
}
```

Launch the app and restart it, it will crash. In my case when code is used in the huge app with other calculations it crashes instantly. In test app i have to run the code twice. With win32 3.1.4 there are no crashes

**Comment by halildurmus** [comment]
Since `v4.0.0` COM objects' pointers are automatically freed by `NativeFinalizer` as mentioned in the [CHANGELOG](https://github.com/dart-windows/win32/blob/main/CHANGELOG.md#400). 
So you need to remove this `calloc.free(ppkf);`

**Comment by kirill-21** [comment]
> Since `v4.0.0`, COM objects' pointers are automatically freed by `NativeFinalizer` as mentioned in the [CHANGELOG](https://github.com/dart-windows/win32/blob/main/CHANGELOG.md#400). So you need to remove this `calloc.free(ppkf);`
> 
> Perhaps we should update the `CHANGELOG` to better explain this with examples.

But there is no .release method for them

**Comment by halildurmus** [comment]
> > Since `v4.0.0`, COM objects' pointers are automatically freed by `NativeFinalizer` as mentioned in the [CHANGELOG](https://github.com/dart-windows/win32/blob/main/CHANGELOG.md#400). So you need to remove this `calloc.free(ppkf);`
> > Perhaps we should update the `CHANGELOG` to better explain this with examples.
> 
> But there is no .release method for them

By COM objects, I mean the classes that extend the `IUnknown` class. In your example, the variable `knownManager` is a COM object and has a `.release()` method.

Note that, with `v5.0.0`, `.release()` is also called automatically if you wish to upgrade it.

**Comment by kirill-21** [comment]
> > > Since `v4.0.0`, COM objects' pointers are automatically freed by `NativeFinalizer` as mentioned in the [CHANGELOG](https://github.com/dart-windows/win32/blob/main/CHANGELOG.md#400). So you need to remove this `calloc.free(ppkf);`
> > > Perhaps we should update the `CHANGELOG` to better explain this with examples.
> > 
> > 
> > But there is no .release method for them
> 
> By COM objects, I mean the classes that extend the `IUnknown` class. In your example, the variable `knownManager` is a COM object and has a `.release()` method. So you can just call `knownManager.release();`.
> 
> Note that, with `v5.0.0`, `.release()` is also called automatically if you wish to upgrade it.

What is the purpose of this update? Seems strange to make it this way

**Comment by kirill-21** [comment]
This is a simple pointer, why making different releasing behavior for same pointer objects? This will create a lot of confuses and misunderstandings

**Comment by halildurmus** [comment]
It's definitely not a simple pointer. `Pointer<COMObject>` is only used by COM objects and we wanted to simplify using COM objects by relieving the burden of manual memory management from developers. 

This helps reduce the chances of accidental resource leaks and makes the code more robust by ensuring that resources are properly released even if developers forget to explicitly call `.release()` and free its pointer.

**Comment by kirill-21** [comment]
> It's definitely not a simple pointer. `Pointer<COMObject>` is only used by COM objects and we wanted to simplify using COM objects by relieving the burden of manual memory management from developers.
> 
> This helps reduce the chances of accidental resource leaks and makes the code more robust by ensuring that resources are properly released even if developers forget to explicitly call `.release()` and free its pointer.
> 
> EDIT: With `v5.0.0`, COM objects now have `.detach()` method that you can call to opt out of this behavior.

This is bad when behavior for same allocation is different, you never know what you should calloc.free by your own and what you must never touch

**Comment by halildurmus** [comment]
> > It's definitely not a simple pointer. `Pointer<COMObject>` is only used by COM objects and we wanted to simplify using COM objects by relieving the burden of manual memory management from developers.
> > This helps reduce the chances of accidental resource leaks and makes the code more robust by ensuring that resources are properly released even if developers forget to explicitly call `.release()` and free its pointer.
> > EDIT: With `v5.0.0`, COM objects now have `.detach()` method that you can call to opt out of this behavior.
> 
> This is bad when behavior for same allocation is different, you never know what you should calloc.free by your own and what you must never touch

This is mentioned in CHANGELOG and also in the [COM documentation](https://win32.pub/docs/com-programming/basics#releasing-com-objects).
